interdiff_test.go

```diff
@@ -18,6 +19,20 @@ var interDiffFileTests = []struct {
 	{"test_examples/result_2.txt", "test_examples/diff_2_a.txt", "test_examples/diff_2_b.txt"},
 }
 
+var applyDiffFileTests = []struct {
+	sourceFile      string
+	diffFile        string
+	resultFile      string
+	finishedWithErr bool
```

Comment:
```suggestion
	wantErr bool
```

The convention for test case terminology is to use "got" and "want". When throwing test failures you should also incorporate both, "got" first, then "want". Something like this:

```
t.Errorf("Failed to apply diff %q: got error: %v; want error: nil", tt.diffFile, err)
```

You should also do that for the rest of the test cases, but no need to do it in this PR.

Reply:
Rest test cases will be corrected in another PR.